Assets/Scripts/Player/Controller.cs

```diff
@@ -290,5 +290,46 @@ private void OnTriggerExit2D(Collider2D other)
         }
 
         #endregion
+        
+        #region Private variables
+        
+        [Range(0, 10)]
+        private int hitPoints = 10;
+
+        /// <summary>
+        /// Events listened by the HUD to update the HUD health bar. Delegates allows to pass variable using events.
+        /// </summary>
+
+        public delegate void UpdateHUDHealthBarHandler(int hitPoints);
+        public event UpdateHUDHealthBarHandler UpdateHealthBarHUD;
+        public event UpdateHUDHealthBarHandler ResetHealthBarHUD;
+        
+        #endregion
+        
+        #region Public Methods
+
+        public void LoseHitPoints(int value)
+        {
+            if (hitPoints - value < 0) hitPoints = 0;
+            else hitPoints -= value;
+            
+            UpdateHealthBarHUD?.Invoke(hitPoints);
+        }
+        
+        public void RestoreHitPoints(int value)
+        {
+            if (hitPoints + value > 10) hitPoints = 10;
+            else hitPoints += value;
+
+            UpdateHealthBarHUD?.Invoke(hitPoints);
+        }
+
+        public void ResetHitPoints()
+        {
+            hitPoints = 10;
+            ResetHealthBarHUD?.Invoke(hitPoints);
+        }
+        
+        #endregion
```

Comment:
Could you make sure these are at the correct positions in the file? There's already a private/public variables&methods further up the file

Reply:
done